Lib/asyncio/locks.py

```diff
@@ -269,17 +269,22 @@ async def wait(self):
                 self._waiters.remove(fut)
 
         finally:
-            # Must reacquire lock even if wait is cancelled
-            cancelled = False
+            # Must reacquire lock even if wait is cancelled.
+            # We only catch CancelledError here, since we don't want any
+            # other (fatal) errors with the future to cause us to spin.
```

Comment:
So interesting that here you don't want to catch other exceptions. (I'm not sure I want to know why. :-)

Reply:
Yes, I thought a lot about that.  It is very rare that one _retries_ an operation after getting an error.  I worry that in case something else bad happen, we might end up in a retry loop.   I wanted to limit the scope of catching errors to a certain class of exceptions here, to avoid running the risk of hanging in case of something really unexpected.  (This is not just some paranoid fantasy, back in my Eve days, we regularly got very mysterious errors seemingly without cause).

The alternative seems to be either limiting the number of loops, which feels dodgy, or limiting it to a class of errors which we know are manually caused.    in the context of my library, I've therefore created a subclass of CancelledError, called InterruptError, to serve as a basis for such signals.